src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -841,8 +857,34 @@ public bool IsTrustedLocation(string path)
             {
                 string location = PathHelper.ValidateDirectory(path);
 
+                var locationDir = new DirectoryInfo(location);
                 // All subdirectories are considered trusted if the parent directory is trusted.
-                return TrustedLocations.FirstOrDefault(x => location.StartsWith(x)) != null;
+                return TrustedLocations.FirstOrDefault(x =>
+                {
+                    if (location.Equals(x))
+                    {
+                        return true;
+                    }
+
+                    if (location.StartsWith(x))
+                    {
+                        var parent = locationDir.Parent;
+                        while(parent != null) 
```

Comment:
What sanity check were you thinking of ?

Reply:
trying this on some nested paths to illustrate that it works and does not hang.

Reply:
I can add some more test cases
locationDir cannot be null because the path that it represents is verified beforehand...but I can add a check for clarity at least